FirebaseABTesting/Sources/FIRExperimentController.m

```diff
@@ -291,4 +291,51 @@ - (NSTimeInterval)latestExperimentStartTimestampBetweenTimestamp:(NSTimeInterval
   }
   return timestamp;
 }
+
+- (void)validateRunningExperimentsForServiceOrigin:(NSString *)origin
+                         runningExperimentPayloads:(NSArray<ABTExperimentPayload *> *)payloads {
+  ABTConditionalUserPropertyController *controller =
+      [ABTConditionalUserPropertyController sharedInstanceWithAnalytics:_analytics];
+
+  FIRLifecycleEvents *lifecycleEvents = [[FIRLifecycleEvents alloc] init];
+
+  // Get the list of experiments from Firebase Analytics.
+  NSArray<NSDictionary<NSString *, NSString *> *> *activeExperiments =
+      [controller experimentsWithOrigin:origin];
+
+  NSMutableSet *runningExperimentIDs = [NSMutableSet setWithCapacity:payloads.count];
+  for (ABTExperimentPayload *payload in payloads) {
+    [runningExperimentIDs addObject:payload.experimentId];
+  }
+
+  for (NSDictionary<NSString *, NSString *> *activeExperimentDictionary in activeExperiments) {
+    NSString *experimentID = activeExperimentDictionary[@"name"];
+    if (![runningExperimentIDs containsObject:experimentID]) {
+      NSString *variantID = activeExperimentDictionary[@"value"];
+
+      [controller clearExperiment:experimentID
+                        variantID:variantID
+                       withOrigin:origin
+                          payload:nil
+                           events:lifecycleEvents];
+    }
+  }
```

Comment:
For code like this it may be better to make a lightweight `NSCopying` Experiment class. Then this method body would be more readable:

```objc
NSArray<FIAMExperiment *> *experiments = /* ... */

// ...

for (FIAMExperiment *experiment in experiments) {
  if (![runningExperimentIDs containsObject:experiment.identifier]) {
    [controller clearExperiment:experiment withOrigin:origin payload:nil events:lifecycleEvents];
  }
}
```

Reply:
Even though FIAM is the only SDK that uses this method currently, I'd like to keep the methods in the ABT SDK generic. I'll probably refactor it down the road because right now it only consists of methods that RC uses and methods that FIAM uses.